### ISSUE TITLE
Can't lose focus from Mui DesktopDatePicker field when clicking on another field

### ISSUE THREAD

**Issue by Selim-ElChoueiter** [mui/mui-x#6159]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [x] I have tested the latest version

### Current behavior 😯

So I have two DesktopDatePickers and I want that when I click anywhere on the field, it opens the Date Picker. However, when I click on one of the fields then click on the other one, the previous one is focused.

### Expected behavior 🤔

Clicking anywhere on a DesktopDatePicker field opens the Date Picker then clicking anywhere else or on any other field closes the Date Picker and focuses the new Field instead of the old one.

### Steps to reproduce 🕹

Link to live example:
https://codesandbox.io/s/frosty-cherry-4cp1sx?file=/src/App.js
Steps:
1. Click on any of the 2 DesktopDatePickers
2. Without selecting a value click on the other DesktopDatePicker

### Context 🔦

I am trying to make all DesktopDatePickers in my page open the DatePicker when I click anywhere on them and not restricted to the Icon on the right (each DatePicker opens when I click on its DesktopDatePicker and closes when losing focus or focusing another DesktopDatePicker).  

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
 System:
    OS: Windows 10 10.0.19044
  Binaries:
    Node: 14.16.0 - ~\node.EXE
    Yarn: 1.22.11 - ~\yarn.CMD
    npm: 6.14.11 - ~\npm.CMD
  Browsers:
    Chrome: 105.0.5195.102
    Edge: Spartan (44.19041.1266.0), Chromium (105.0.1343.33)
  npmPackages:
    @emotion/react: 11.6.0 => 11.6.0
    @emotion/styled: 11.6.0 => 11.6.0
    @mui/core:  5.0.0-alpha.54
    @mui/icons-material: 5.1.0 => 5.1.0
    @mui/lab: ^5.0.0-alpha.54 => 5.0.0-alpha.54
    @mui/material: 5.1.0 => 5.1.0
    @mui/private-theming:  5.1.0
    @mui/styled-engine:  5.1.0
    @mui/styles: ^5.1.0 => 5.1.0
    @mui/system:  5.1.0
    @mui/types:  7.1.0
    @mui/utils:  5.1.0
    @mui/x-date-pickers: ^5.0.0-beta.1 => 5.0.0-beta.1
    @types/react:  17.0.24
    react: 17.0.2 => 17.0.2
    react-dom: 17.0.2 => 17.0.2
```
</details>

**Comment by alexfauquette** [comment]
I'm a bit concerned by your goal on the aspect of focus management.

When you click on the input, where should be the focus? Because Right now the calendar is taking the focus, it prevents using the keyboard to write the date.

If you want to force using the calendars, you can simply use the MobileDatePickers

**Comment by Selim-ElChoueiter** [comment]
@alexfauquette The problem is we have used these date picker through our website but only need it changed in one page. Is there a way to not allow focus on the field but be able to display the calendar and pick a date from it? (I tried the readonly prop but this also blocks the date picked by the calendar) Or should we change the component to a different date picker (in this case the MobileDatePicker)?

**Comment by alexfauquette** [comment]
For me, the easiest is to use `MobileDatePicker` the only drawback is that you will get a `Modal` instead of a `Popper`.

About the bug itself, I assume it's linked to the timeout used for moving back to the last focused element:
https://github.com/alexfauquette/mui-x/blob/keyboard-column-group/packages/x-date-pickers/src/internals/components/PickersPopper.tsx#L249:L268

**Comment by Selim-ElChoueiter** [comment]
I think this might be causing it, is there a workaround to get past it? If not I guess I will have to take a look at the MobileDatePicker component.

**Comment by alexfauquette** [comment]
I do not think this is overridable. For deep customization of the open/close state, the easiest is probably to redo this aspect of the component by yourself.

Do to so you can use `StaticPickers` which are in fact the interactive interface.

Here is a codesabdbox with pickers displayed in a popper or a popover with a button instead of the text field
https://codesandbox.io/s/crazy-bird-0kvx1n?file=/demo.tsx

**Comment by Selim-ElChoueiter** [comment]
We will definitely look into it! Thank you very much!

**Comment by alexfauquette** [comment]
I reopen this issue, to keep track of the progress we can do.

I think we could improve the focus management because this `setTimeout` is not fully satisfying. Or at least document how to do a custom component based on static ones as proposed in the previous comment

**Comment by alexfauquette** [comment]
I'm closing since it seems to now work with to following adaptation for v6

```jsx
slotProps={{
  textField: {
    inputRef: createdToRef,
    onClick: () => {
      setOpenCreatedTo((prev) => !prev);
      if (createdToRef && createdToRef.current !== null) {
        createdToRef.current.blur();
      }
    }
  }
}}
```

https://codesandbox.io/s/amazing-brahmagupta-fwht9v?file=/src/App.js

**Comment by westanvv** [comment]
I'd faced the same problem in v6. I had a controlled open state. My solution:
```
slotProps={{
  popper: {
    onBlur: () => {
      // update state
    },
  },
}}
```
